lua/lualine/components/diagnostics/init.lua

```diff
@@ -104,13 +104,18 @@ function M:update_status()
 
   -- format the counts with symbols and highlights
   if self.options.colored then
-    local colors = {}
+    local colors, bgs = {}, {}
     for name, hl in pairs(self.highlight_groups) do
       colors[name] = modules.highlight.component_format_highlight(hl)
+      bgs[name] = self.options.diagnostics_color[name].bg or nil
     end
+    local padding
+    local previous_section = nil
```

Comment:
You can declare both padding & previous section in same line.

Reply:
luacheck didn't "liked" it:
```lua
local previous_section, padding = nil, nil -- Complains about the value never being used since padding is redefined
local previous_section, padding = nil -- Complains that a value is missing and nil will be used instead
```

Reply:
If you don't assign it'll be set to nil.

You just need .
```lua
local previous_section, padding
```

Reply:
Good to know ;)